+++
# Imported from original pyblosxom .txt format at 2015-08-14
date = "2006-09-10T06:29:42-07:00"
title = "Error running scapy as non-root\n"
# Marked a draft because frankly my old writing may not be worth surfacing again.
# I made bad word choices and had some strong-and-closed opinions years ago.
# We learn over time, eh? :)
draft = true
type = "old"
categories = [ "old" ]
tags = ["scapy", "python"]
+++

If you get this error:
<pre>
Traceback (most recent call last):
  File "/usr/local/bin/scapy", line 10647, in ?
    class Conf(ConfClass):
  File "/usr/local/bin/scapy", line 10670, in Conf
    iface = get_working_if()
  File "/usr/local/bin/scapy", line 2067, in get_working_if
    except pcap.pcapc.EXCEPTION:
AttributeError: 'module' object has no attribute 'pcapc'
</pre>

This is becuase you aren't running scapy as root. Run it as root.

<p>

If you still get this error message, it's likely due to pcap failing to find usable network interfaces. This means you have no interfaces in the UP state. It doesn't count lo0 as a real interface, I guess?
